lib/associations/belongs-to-many.js

```diff
@@ -302,6 +304,71 @@ var BelongsToMany = function(source, target, options) {
 
 util.inherits(BelongsToMany, Association);
 
+BelongsToMany.prototype.get = function(instances, options) {
+  var throughModel = this.throughModel;
+
+  var throughOptions = _.assign({
+    where: _.set({}, this.foreignKey, {$in: _.map(instances, this.source.primaryKeyAttribute)}),
+    logging: options.logging
+  });
+
+  var association = this;
+
+  if(_.isUndefined(options.limit) && _.isUndefined(options.orderBy)) {
+    return throughModel.findAll(throughOptions).then(function (throughs) {
+      var otherKey = association.otherKey
+          , targetIds = _.chain(throughs).map(otherKey).uniq().value();
+
+      var targetOptions = _.chain({
+            where: _.set({}, association.target.primaryKeyAttribute, {$in: targetIds})
+          })
+          .assign(association.target.$optClone(options))
+          .omit('include', 'includeMap', 'includeNames')
+          .value();
+
+      return Utils.Promise.join(throughs, _.isEmpty(targetIds) ? null : association.target.findAll(targetOptions));
+    }).then(_.spread(function (throughs, targets) {
+      targets = _.groupBy(targets, association.target.primaryKeyAttribute);
+
+      return _.chain(instances)
+          .map(association.source.primaryKeyAttribute)
+          .map(function (sourceId) {
+            var instanceThroughs = _.filter(throughs, function (through) {
+              return through.get(association.foreignKey, {raw: true}) === sourceId;
+            })
+                , instanceTargets = _.at(targets, _.map(instanceThroughs, association.otherKey));
+
+            return [sourceId, _(instanceTargets).compact().flatten().value()];
+          })
+          .fromPairs()
+          .value();
+    }));
+  } else {
+    return Utils.Promise.resolve(instances)
+      .map(function (instance) {
+        var throughOptions = _.assign(
+          {
+            include: [
+              {
+                association: association.target.hasOne(throughModel),
+                where: _.set({}, association.foreignKey, instance.get(association.source.primaryKeyAttribute))
+              }
+            ]
+          },
+          _.pick(options, 'order', 'limit', 'logging')
```

Comment:
we want users to be able to pass a top level options object that makes it all the way down regardless.
this is helpfull to plugins, performance tracing etc